src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -362,24 +395,32 @@ public class BlobFileStore : IFileStore
             Conditions = _blobClient.GetConditions(fileProperties),
         };
 
-        Response<BlobDownloadResult> result = await ExecuteAsync(
-            action: async () => await blob.DownloadContentAsync(blobDownloadOptions, cancellationToken),
+        BlobDownloadResult result = await ExecuteAsync(
+            func: async () =>
+            {
+                Response<BlobDownloadResult> result = await blob.DownloadContentAsync(blobDownloadOptions, cancellationToken);
```

Comment:
If you did want to remove the code block, you could rely on the [`implicit` conversion operation of `Response<T>` to `T`](https://github.com/Azure/azure-sdk-for-net/blob/c284046de29a1561373843a0f4f33600c474348a/sdk/core/Azure.Core/src/ResponseOfT.cs#L32). You'll need to provide a hint to the generic method, but I think this should compile:

```csharp
BlobDownloadResult result = await ExecuteAsync<BlobDownloadResult>(
    func: async () => await blob.DownloadContentAsync(blobDownloadOptions, cancellationToken),
    operationName: nameof(GetFileContentInRangeAsync),
    operationType: OperationType.Output,
    extractLength: long? (result) => result.Details.ContentLength);
```